velox/connectors/fuzzer/tests/FuzzerConnectorTest.cpp

```diff
@@ -0,0 +1,163 @@
+        connector::getConnectorFactory(
+            connector::fuzzer::FuzzerConnectorFactory::kFuzzerConnectorName)
+            ->newConnector(kFuzzerConnectorId, nullptr);
+    connector::registerConnector(fuzzerConnector);
+  }
+
+  void TearDown() override {
+    connector::unregisterConnector(kFuzzerConnectorId);
+    OperatorTestBase::TearDown();
+  }
+
+  exec::Split makeFuzzerSplit(size_t numRows) const {
+    return exec::Split(
+        std::make_shared<FuzzerConnectorSplit>(kFuzzerConnectorId, numRows));
+  }
+
+  std::vector<exec::Split> makeFuzzerSplits(
+      size_t rowsPerSplit,
+      size_t numSplits) const {
+    std::vector<exec::Split> splits;
+    splits.reserve(numSplits);
+
+    for (size_t i = 0; i < numSplits; ++i) {
+      splits.emplace_back(makeFuzzerSplit(rowsPerSplit));
+    }
+    return splits;
+  }
+
+  std::shared_ptr<FuzzerTableHandle> makeFuzzerTableHandle(
+      size_t fuzzerSeed = 0) const {
+    return std::make_shared<FuzzerTableHandle>(
+        kFuzzerConnectorId, fuzzerOptions_, fuzzerSeed);
+  }
+
+ private:
+  VectorFuzzer::Options fuzzerOptions_;
+};
+
+TEST_F(FuzzerConnectorTest, singleSplit) {
+  const size_t numRows = 100;
+  auto type = ROW({BIGINT(), DOUBLE(), VARCHAR()});
+
+  auto plan =
+      PlanBuilder().tableScan(type, makeFuzzerTableHandle(), {}).planNode();
+
+  exec::test::AssertQueryBuilder(plan)
+      .split(makeFuzzerSplit(numRows))
+      .assertNumRows(type, numRows);
+}
+
+TEST_F(FuzzerConnectorTest, complexTypes) {
+  const size_t numRows = 100;
+  auto type = ROW({
+      ARRAY(BIGINT()),
+      ROW({VARCHAR(), MAP(INTEGER(), ARRAY(DOUBLE())), VARBINARY()}),
+      REAL(),
+  });
+
+  auto plan =
+      PlanBuilder().tableScan(type, makeFuzzerTableHandle(), {}).planNode();
+
+  exec::test::AssertQueryBuilder(plan)
+      .split(makeFuzzerSplit(numRows))
+      .assertNumRows(type, numRows);
+}
+
+TEST_F(FuzzerConnectorTest, multipleSplits) {
+  const size_t rowsPerSplit = 100;
+  const size_t numSplits = 10;
+  auto type = ROW({BIGINT(), DOUBLE(), VARCHAR()});
+
+  auto plan =
+      PlanBuilder().tableScan(type, makeFuzzerTableHandle(), {}).planNode();
+
+  exec::test::AssertQueryBuilder(plan)
+      .splits(makeFuzzerSplits(rowsPerSplit, numSplits))
+      .assertNumRows(type, rowsPerSplit * numSplits);
+}
+
+TEST_F(FuzzerConnectorTest, randomTypes) {
+  const size_t rowsPerSplit = 100;
+  const size_t numSplits = 10;
+
+  const size_t iterations = 20;
+
+  for (size_t i = 0; i < iterations; ++i) {
+    auto type = VectorFuzzer({}, pool()).randRowType();
+
+    auto plan =
+        PlanBuilder().tableScan(type, makeFuzzerTableHandle(), {}).planNode();
+    exec::test::AssertQueryBuilder(plan)
+        .splits(makeFuzzerSplits(rowsPerSplit, numSplits))
+        .assertNumRows(type, rowsPerSplit * numSplits);
+  }
+}
+
+TEST_F(FuzzerConnectorTest, reproducible) {
+  const size_t numRows = 100;
+  auto type = ROW({BIGINT(), ARRAY(INTEGER()), VARCHAR()});
```

Comment:
nit: Could we add a floating-point child type here? I'm curious whether our test utility function `exec::test::assertEqualResults` works with the random data.

Reply:
Good point. Let me try to add another test case